File: rust/ppm.rs
// Program creates ppm image from given pixels data

use std::fs::File;
use std::io;
use std::io::Write;

fn save_as_ppm(file_path: &str, pixels: &[u32], height: usize, width: usize) -> io::Result<()>
{
    let mut file = File::create(file_path)?;
    write!(file, "P6\n{} {} 255\n", width, height);
    for y in 0..width
    {
        for x in 0..height
        {
            let pixel = pixels[y * width + x];
            let color = [((pixel >> 16) & 0xFF) as u8,
                            ((pixel >> 8) & 0xFF) as u8,
                            (pixel & 0xFF) as u8
            ];
            print!("{:?}", color);
            file.write(&color)?;
        }
    }
    Ok(())
}

fn main()
{
    const WIDTH: usize = 64;
    const HEIGHT: usize = 64;
    const OUTPUT_PATH: &str = "output.ppm";
    let mut pixels = [0u32; WIDTH * HEIGHT];
    pixels.fill(0xFFFFFF);
    pixels[0] = 0x000010;
    pixels[1] = 0x000010;
    pixels[2] = 0x000010;
    pixels[3] = 0x000010;
    pixels[4] = 0x000010;
    pixels[5] = 0x000012;
    pixels[6] = 0x100010;
    pixels[7] = 0x011100;
    pixels[8] = 0x000000;
    pixels[9] = 0x000000;
    save_as_ppm(OUTPUT_PATH, &pixels, HEIGHT, WIDTH);
}

File: py/peanut/Lib/ppm.py
#!/usr/bin/env python3

import os
import sys

def save_as_ppm(path, pixels, height, width) -> bool:
    with open(path, "wb") as ppm_out:
        ppm_out.write(f'P6\n{width} {height}\n255\n'.encode())
        for y in range(width):
            for x in range(height):
                pixel = pixels[y * width + x]
                color = bytearray([(pixel >> 16) & 0xFF, (pixel >> 8) & 0xFF, pixel & 0xFF])
                ppm_out.write(color)

def main():
    import random
    H = 256
    W = 256
    pixels = []
    for y in range(W):
        for x in range(H):
            pixel = (random.randint(0, 255) << 16) | (random.randint(0, 255) << 8) | (random.randint(0, 255) << 0)
            pixels.append(pixel)

    save_as_ppm("output.ppm", pixels, H, W)

if __name__ == "__main__":
    main()
